crates/libcontainer/src/workload/mod.rs

```diff
@@ -0,0 +1,77 @@
+use anyhow::{Context, Result};
+use oci_spec::runtime::Spec;
+
+use self::default::DefaultExecutor;
+#[cfg(feature = "wasm-wasmer")]
+use self::wasmer::WasmerExecutor;
+
+pub mod default;
+#[cfg(feature = "wasm-wasmer")]
+pub mod wasmer;
+
+static EMPTY: Vec<String> = Vec::new();
+
+pub trait Executor {
+    /// Executes the workload
+    fn exec(&self, spec: &Spec) -> Result<()>;
+    /// Checks if the handler is able to handle the workload
+    fn can_handle(&self, spec: &Spec) -> Result<bool>;
+    /// The name of the handler
+    fn name(&self) -> &str;
+}
+pub struct CompositeExecutor {
+    executors: Vec<Box<dyn Executor>>,
+}
+
+impl Executor for CompositeExecutor {
```

Comment:
What's the benefit for having a CompositeExecutor? It seems the major benefit for an Executor is that is creates a common interface for Executors, could this maybe just be a ExecutorManager that doesn't need to satisfy `Executor`? I don't imagine we'll ever need the `name` function here, and that's just dead code.